prow/pjutil/help.go

```diff
@@ -58,29 +58,38 @@ func ShouldRespondWithHelp(body string, toRunOrSkip int) (bool, string) {
 	}
 }
 
-func HelpMessage(org, repo, branch, note string, testAllNames, testCommands []string) string {
+func HelpMessage(org, repo, branch, note string, testAllCommands, optionalTestCommands, requiredTestCommands []string) string {
 	var resp string
-	if len(testAllNames)+len(testCommands) > 0 {
-		listBuilder := func(names []string) string {
-			var list strings.Builder
-			for _, name := range names {
-				list.WriteString(fmt.Sprintf("\n* `%s`", name))
-			}
-			return list.String()
-		}
+	if len(testAllCommands)+len(optionalTestCommands)+len(requiredTestCommands) == 0 {
+		return fmt.Sprintf("No presubmit jobs available for %s/%s@%s", org, repo, branch)
+	}
 
-		var testAllNote string
-		if len(testAllNames) == len(testCommands) {
-			testAllNote = "Use `/test all` to run all jobs.\n"
-		} else if len(testAllNames) > 0 {
-			testAllNote = fmt.Sprintf("Use `/test all` to run the following jobs:%s\n\n", listBuilder(testAllNames))
+	listBuilder := func(names []string) string {
+		var list strings.Builder
+		for _, name := range names {
+			list.WriteString(fmt.Sprintf("\n* `%s`", name))
 		}
+		return list.String()
+	}
 
-		resp = fmt.Sprintf("%sThe following commands are available to trigger jobs:%s\n\n%s",
-			note, listBuilder(testCommands), testAllNote)
-	} else {
-		resp = fmt.Sprintf("No presubmit jobs available for %s/%s@%s", org, repo, branch)
+	var testAllNote string
+	if len(testAllCommands) == len(optionalTestCommands)+len(requiredTestCommands) {
+		testAllNote = "Use `/test all` to run all jobs.\n"
+	} else if len(testAllCommands) > 0 {
+		testAllNote = fmt.Sprintf("Use `/test all` to run the following jobs:%s\n\n", listBuilder(testAllCommands))
```

Comment:
nit: seems like this could be moved below the `if len(optionalTestCommands) > 0` block below, just above the `resp += testAllNote`